Lib/test/test_heapq.py

```diff
@@ -135,6 +153,13 @@ def test_heappushpop(self):
         x = self.module.heappushpop(h, 11)
         self.assertEqual((h, x), ([11], 10))
 
+    def test_heappop_max(self):
```

Comment:
It's https://github.com/python/cpython/blob/43ba8861e0ad044efafa46a7cc04e12ac5df640e/Lib/heapq.py#L187 (the return statement used when the heap is now empty and doesn't need rebalancing).

I did wonder if I could test this via `merge`, but it looks like `merge` has an optimisation where it never calls `heappop_max` on a one-item list (https://github.com/python/cpython/blob/43ba8861e0ad044efafa46a7cc04e12ac5df640e/Lib/heapq.py#L354).

I'll change this test to be called on `self.module`, but I can't see a way to hit that line of Python code by just going through the public API (though I could refactor it to only have a single return statement?).